### PR TITLE
[ARM] Add az feature unregister api

### PR DESCRIPTION
**Pull request by diwudd**
Add az feature unregister api
Issue Raised: https://github.com/Azure/azure-cli/issues/8941

### CODE REVIEW

**Comment by yonzhan** [comment]
add to S170

**Comment by yonzhan** [comment]
add to S171

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py [hunk 1]
```diff
@@ -1130,6 +1130,10 @@ def test_feature_list(self):
         # Once a feature goes GA , it will be removed from the feature list. Once that happens, use other ones to test
         self.cmd('feature show --namespace Microsoft.Network -n AllowLBPreview')
 
+    @AllowLargeResponse(8192)
+    def test_feature_unregister(self):
+        self.cmd('feature unregister --namespace Microsoft.Network --name AllowLBPreview', checks=self.check('properties.state', 'Unregistered'))
```
Comment (haroldrandom) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1135]: Should the test need pre-setup step like register Network namespace before running this?

Comment (diwudd) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1135]: No, it does not need pre-setup

**Comment by diwudd** [comment]
Thanks Jianhui.

@Juliehzl,   can you please take another look?

Thanks,

Di

**Comment by ifeoluwaokunoren** [comment]
@Juliehzl can you please confirm your comments have been addressed, and approve so we can merge?
Thanks.

File: src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py (hunk 1 above)
Comment (Juliehzl) [inline src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1135]: Can we add some check for command response? Or it is possible to utilize `az feature show` to check state after unregistering?

Comment (diwudd) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1135]: I might miss something. It does check the response with "checks=self.check('properties.state', 'Unregistered')". Anything else you think we should check here?

Comment (Juliehzl) [reply src/azure-cli/azure/cli/command_modules/resource/tests/latest/test_resource.py:1135]: yes. you can check the state in response.

**Pull request merged**